clients/src/test/java/org/apache/kafka/clients/consumer/RoundRobinAssignorTest.java

```diff
@@ -181,6 +172,114 @@ public void testTwoConsumersTwoTopicsSixPartitions() {
         assertEquals(partitions(tp(topic1, 1), tp(topic2, 0), tp(topic2, 2)), assignment.get(consumer2));
     }
 
+    @Test
+    public void testTwoStaticConsumersTwoTopicsSixPartitions() {
```

Comment:
Could we add a unit test that a static member have changed its member.id (due to a fail-over, e.g.) while maintaining its instance.id, and all other members stay the same; and as a result it still get the same assignment?